src/libs/vtkh/filters/UniformGrid.cpp

```diff
@@ -323,13 +336,13 @@ class GlobalReduceField
         {
           for(int i = 0; i < num_points; ++i)
           {
-            if(g_mask[i] == 1)
-            {
-                    global_field[i] = m_invalid_value;
-            }
-            if(g_valid[i] > 1)
+            if(g_rank_mask[i] == -1)
             {
-                    global_field[i] = global_field[i]/g_valid[i];
+              global_field[i] = m_invalid_value;
+#if _DEBUG 
+              global_field[i] = par_rank*10;
+#endif
+	      std::cerr << "INVALID VALUE>>>>>>>>>>>>>>>>>>>>>>>>>" << std::endl;
```

Comment:
same, do we need this cerr?

Reply:
thank you! Thought I caught all of these. We should only have cerr within the _DEBUGs. I'll do another look through as well.